internal/ethapi/api.go

```diff
@@ -1246,6 +1246,71 @@ func (s *PublicTransactionPoolAPI) GetTransactionReceipt(ctx context.Context, ha
 	return fields, nil
 }
 
+// GetBlockReceipts returns the block receipts for the given block number.
+func (s *PublicTransactionPoolAPI) GetBlockReceipts(ctx context.Context, blockNr rpc.BlockNumber) ([]map[string]interface{}, error) {
+	block, err := s.b.BlockByNumber(ctx, blockNr)
+	if err != nil {
+		return nil, err
+	}
+	if block == nil {
+		return nil, nil
+	}
+
+	txs := block.Transactions()
+
+	receipts, err := s.b.GetReceipts(ctx, block.Hash())
+	if err != nil {
+		return nil, err
+	}
+
+	if len(receipts) != len(txs) {
+		return nil, fmt.Errorf("receipt and transaction count mismatch")
+	}
+
+	result := make([]map[string]interface{}, 0, len(receipts))
+
+	for index, receipt := range receipts {
```

Comment:
receipt.TransactionIndex could stand for index, so `index` here makes no sense. Suggest to replace it with blank identifier `"_"`